gpkit/constraints/model.py

```diff
@@ -134,6 +136,35 @@ def _add_modelname_tovars(self, name, num):
         with SignomialsEnabled():  # since we're just substituting varkeys.
             self.subinplace(add_model_subs)
 
+    def sweep(self, sweeps, **solveargs):
+        "Sweeps variable, returning swept solution"
+        sols = []
+        for sweepvar, sweepvals in sweeps.items():
+            original_val = self.substitutions.get(sweepvar, None)
+            self.substitutions.update({sweepvar: ('sweep', sweepvals)})
+            try:
+                sols.append(self.solve(**solveargs))
+            except InvalidGPConstraint:
+                sols.append(self.localsolve(**solveargs))
+            if original_val:
+                self.substitutions[sweepvar] = original_val
+            else:
+                del self.substitutions[sweepvar]
+        if len(sols) == 1:
+            return sols[0]
```

Comment:
I think most people will be using `.sweep({...})` to do a single 1D sweep, so this makes that a lot simpler.